OpenRA.Game/Graphics/TerrainSpriteLayer.cs

```diff
@@ -227,6 +237,40 @@ public void Dispose()
 				worldRenderer.TerrainLighting.CellChanged -= UpdateTint;
 
 			vertexBuffer.Dispose();
+
+			lock (IndexBuffers)
+				indexBuffer.Dispose();
+		}
+	}
+
+	public sealed class IndexBufferRc : IDisposable
+	{
+		public IIndexBuffer Buffer;
+		int count;
+
+		IndexBufferRc(World world)
+		{
+			var indices = new uint[6 * world.Map.MapSize.X * world.Map.MapSize.Y];
+			var cornerVertexMap = new uint[] { 0, 1, 2, 2, 3, 0 };
+			for (var i = 0; i < indices.Length; i++)
+				indices[i] = cornerVertexMap[i % 6] + (uint)(4 * (i / 6));
+
+			Buffer = Game.Renderer.Context.CreateIndexBuffer(indices.Length);
+			Buffer.SetData(indices, indices.Length);
+		}
+
+		public static IndexBufferRc Create(World w)
```

Comment:
This Create isn't needed I guess (sorry for adding it in the skeleton), just expose the ctor and move everything there.